tests/foreman/api/test_repository.py

```diff
@@ -208,23 +221,26 @@ def test_update(self, attrs):
         @Feature: Repository
 
         """
-        client.put(
-            self.repository.path(),
-            attrs,
-            auth=get_server_credentials(),
-            verify=False,
-        ).raise_for_status()
-        real_attrs = self.repository.read_json()
-        for name, value in attrs.items():
-            self.assertIn(name, real_attrs.keys())
-            if name == 'content_type':
-                # Cannot update a repository's content type.
-                self.assertEqual(
-                    self.repository.get_fields()['content_type'].default,
-                    real_attrs[name]
-                )
-            else:
-                self.assertEqual(value, real_attrs[name])
+        for attrs in _test_data():
+            with self.subTest(attrs):
+                client.put(
```

Comment:
Probably.  But I am reusing the old code as such and just removing ddt logic.  Removing references to client.put should be a separate effort - I created a github issue for this: https://github.com/SatelliteQE/robottelo/issues/2837